reco_utils/evaluation/python_evaluation.py

```diff
@@ -325,63 +352,48 @@ def merge_ranking_true_pred(
     """Filter truth and prediction data frames on common users
 
     Args:
-        rating_true (pd.DataFrame): True data.
-        rating_pred (pd.DataFrame): Predicted data.
-        col_user (str): column name for user.
-        col_item (str): column name for item.
-        col_rating (str): column name for rating.
-        col_prediction (str): column name for prediction.
-
+        rating_true (pd.DataFrame): True DataFrame
+        rating_pred (pd.DataFrame): Predicted DataFrame
+        col_user (str): column name for user
+        col_item (str): column name for item
+        col_rating (str): column name for rating
+        col_prediction (str): column name for prediction
+        relevancy_method (str): method for determining relevancy ['top_k', 'by_threshold']
+        k (int): number of top k items per user (optional)
+        threshold (float): threshold of top items per user (optional)
     Returns:
-        pd.DataFrame: new data frame of true data DataFrame of recommendation hits
-            number of common users
+        pd.DataFrame, pd.DataFrame: DataFrame of recommendation hits, and hit counts vs actual relevant items per user
     """
-    relevant_func = {"top_k": get_top_k_items}
 
-    rating_pred_new = (
-        relevant_func[relevancy_method](
-            dataframe=rating_pred,
-            col_user=col_user,
-            col_rating=col_prediction,
-            threshold=threshold,
-        )
-        if relevancy_method == "by_threshold"
-        else relevant_func[relevancy_method](
-            dataframe=rating_pred, col_user=col_user, col_rating=col_prediction, k=k
-        )
-    )
+    # Make sure the prediction and true data frames have the same set of users
+    common_users = list(set(rating_true[col_user]).intersection(set(rating_pred[col_user])))
+    rating_true_common = rating_true[rating_true[col_user].isin(common_users)]
+    rating_pred_common = rating_pred[rating_pred[col_user].isin(common_users)]
 
-    common_user_list = list(
-        set(rating_true[col_user]).intersection(set(rating_pred_new[col_user]))
-    )
-
-    # To make sure the prediction and true data frames have the same set of users.
-    rating_pred_new = rating_pred_new[rating_pred_new[col_user].isin(common_user_list)]
-    rating_true_new = rating_true[rating_true[col_user].isin(common_user_list)]
-
-    n_users = len(common_user_list)
-
-    # Return hit items in prediction data frame with ranking information. This
-    # is used for calculating
-    # NDCG and MAP.
-    # Use first to generate unique ranking values for each item. This is to align with the
-    # implementation in
-    # Spark evaluation metrics, where index of each recommended items (the indices are unique
-    #  to items) is used
+    # Return hit items in prediction data frame with ranking information. This is used for calculating NDCG and MAP.
```

Comment:
curiosity, was this formatted with black or did you do it manually?

Reply:
i tried to manually clean it up, but black may be fighting with me. I really prefer 120 character line limits...